x-pack/plugins/actions/server/authorization/get_authorization_mode_by_source.ts

```diff
@@ -34,36 +34,57 @@ export async function getAuthorizationModeBySource(
     : AuthorizationMode.RBAC;
 }
 
-export async function getBulkAuthorizationModeBySource(
+export async function bulkGetAuthorizationModeBySource(
+  logger: Logger,
   unsecuredSavedObjectsClient: SavedObjectsClientContract,
   executionSources: Array<ActionExecutionSource<unknown>> = []
-): Promise<Record<string, number>> {
-  const count = { [AuthorizationMode.Legacy]: 0, [AuthorizationMode.RBAC]: 0 };
-  if (executionSources.length === 0) {
-    count[AuthorizationMode.RBAC] = 1;
-    return count;
+): Promise<Record<AuthorizationMode, number>> {
+  const authModes = { [AuthorizationMode.Legacy]: 0, [AuthorizationMode.RBAC]: 0 };
+
+  const alertSavedObjectExecutionSources: SavedObjectExecutionSource[] = executionSources.filter(
+    (source) =>
+      isSavedObjectExecutionSource(source) && source?.source?.type === ALERT_SAVED_OBJECT_TYPE
+  ) as SavedObjectExecutionSource[];
+
+  // If no ALERT_SAVED_OBJECT_TYPE source, default to RBAC
+  if (alertSavedObjectExecutionSources.length === 0) {
```

Comment:
We only need to get the saved object if it is an alert saved object execution source, so adding this check so we can exit early if we can.